Automattic/wp-calypso
# InlineHelp: Add In-modal Support Articles

### Summary

This PR aims to add Support Articles as a rich-result type for inline-help, as opposed to an external link as it is right now.

The strategy for adding this view was to ~steal~ borrow as much as possible from an existing view found in reader -- `blocks/reader-full-post` -- with the idea that this would get this feature working faster with minimal deviations between the style of the inline support articles and the articles found in the readers regular view. A downside of this hasty approach is that there was a lot to copy over and prune out - I've attempted to shave off as much as I could from the copied code but please keep 👀 open for anything that's extraneous.

We decided to move the copied views to the `inline-help` directory so as to not advertise this component(s) as being reusable elsewhere in the app as right now it caters only for our support articles. That said, if it becomes apparent that this component would be useful elsewhere we could then look at pulling it out in to `/components/` and making it a little more robust. 

Another note: we had hoped originally to be able to just borrow styles directly from reader. The trouble there is that readers styles are loaded only when necessary - when the customer is looking at the reader side of WP.com. We opted for avoiding pulling these styles in and breaking down some of the work that's been done to split our payload.

### Test Plan

- Test that articles appear in the dialog as expected
  - Go to http://calypso.localhost:3000/
  - Open InlineHelp by clicking the ? button in the bottom right
    - You should see a result for 'Reader', which leads to a Support  Article
  - Click 'Reader' then the button that shows labeled 'Read More'
    - You should now see the dialog open with a loading state (not finished)
    - The loading state should then be replaced with the article content (styling not finished)

- Test Video Results for regressions
  - Go to http://calypso.localhost:3000/media
    - You should see a result for 'Add a Photo Gallery', which leads to a Support  Article
  - Click 'Add a Photo Gallery' then the button that shows labeled 'Watch Video'
  - The Dialog should show with a video and without regressions

- Test anomalies (learn.wordpress.com / support.wordpress.com)
  - Go to http://calypso.localhost:3000/help
  - Open InlineHelp by clicking the ? button in the bottom right
  - Click "Self-guided online tutorial"
    -  You should be taken directly to `learn.wordpress.com` (Do not pass go...)
  - Click "All support articles"
    -  You should be taken directly to `support.wordpress.com/` (...Do not collect $200)
